internal/devbox/providers/nixcache/nixcache.go

```diff
@@ -2,91 +2,201 @@ package nixcache
-	"go.jetpack.io/pkg/auth"
 	"go.jetpack.io/pkg/filecache"
 )
 
 type Provider struct{}
 
 var singleton *Provider = &Provider{}
 
-type NixCacheConfig struct {
-	URI         string
-	Credentials types.Credentials
+func Get() *Provider {
+	return singleton
 }
 
-func (n NixCacheConfig) CredentialsEnvVars() []string {
-	env := []string{}
-	if n.Credentials.AccessKeyId != nil {
-		env = append(env, "AWS_ACCESS_KEY_ID="+*n.Credentials.AccessKeyId)
+func (p *Provider) ConfigureAWS(ctx context.Context, username string) error {
+	rootConfig, err := p.rootAWSConfigPath()
+	if err != nil {
+		return err
+	}
+	if fileutil.Exists(rootConfig) {
+		// Already configured.
+		return nil
 	}
-	if n.Credentials.SecretKey != nil {
-		env = append(env, "AWS_SECRET_ACCESS_KEY="+*n.Credentials.SecretKey)
+
+	if os.Getuid() == 0 {
+		err := p.configureRoot(username)
+		if err != nil {
+			return redact.Errorf("update ~root/.aws/config with devbox credentials: %s", err)
+		}
+		return nil
 	}
-	if n.Credentials.SessionToken != nil {
-		env = append(env, "AWS_SESSION_TOKEN="+*n.Credentials.SessionToken)
+
+	_, err = nix.DaemonVersion(ctx)
+	if err == nil {
+		// It looks like this is a multi-user install running a Nix
+		// daemon, so we need to configure AWS S3 authentication for the
+		// root user.
+		if err := p.sudoConfigureRoot(ctx, username); err != nil {
+			return err
+		}
 	}
-	return env
+	return nil
 }
 
-func Get() *Provider {
-	return singleton
+func (p *Provider) rootAWSConfigPath() (string, error) {
+	u, err := user.LookupId("0")
+	if err != nil {
+		return "", redact.Errorf("lookup root user: %s", err)
+	}
+	if u.HomeDir == "" {
+		return "", redact.Errorf("empty root user home directory: %s", u.Username, err)
+	}
+	return filepath.Join(u.HomeDir, ".aws", "config"), nil
 }
 
-// Config returns the URI or the nix bin cache and AWS credentials if available.
-// Nix calls the URI a substituter.
-// A substituter is a bin cache URI that nix can use to fetch pre-built
-// binaries from.
-func (p *Provider) Config(ctx context.Context) (NixCacheConfig, error) {
-	token, err := identity.Get().GenSession(ctx)
-
-	if errors.Is(err, auth.ErrNotLoggedIn) {
-		// DEVBOX_NIX_BINCACHE_URI seems like a friendlier name than "substituter"
-		return NixCacheConfig{
-			URI: os.Getenv("DEVBOX_NIX_BINCACHE_URI"),
-		}, nil
-	} else if err != nil {
-		return NixCacheConfig{}, err
-	}
-
-	apiClient := api.NewClient(ctx, build.JetpackAPIHost(), token)
-	cache := filecache.New[*nixv1alpha1.GetBinCacheResponse]("devbox/credentials")
-	binCacheResponse, err := cache.GetOrSetWithTime(
-		"aws-nix-bin-cache",
-		func() (*nixv1alpha1.GetBinCacheResponse, time.Time, error) {
-			r, err := apiClient.GetBinCache(ctx)
-			if err != nil || r.GetNixBinCacheUri() == "" {
-				return nil, time.Time{}, err
-			}
-			return r, r.GetNixBinCacheCredentials().Expiration.AsTime(), nil
-		},
-	)
+func (p *Provider) configureRoot(username string) error {
+	exe, err := os.Executable()
```

Comment:
Keep in mind that prod devbox is wrapped. Calling unwrapped version may cause issues.

ditto on other line that looks up executable. 

I think for this feature (and just to be safe) we enforce that executable is just `devbox`? 

I remember using the unwrapped version at some point in our bin wrappers and it caused all sorts of issues (specifically, the wrapper sets some env vars that the binary expects I think).

Reply:
root might not have devbox in its PATH like the user does. I'll see if there's a way to get the path to the wrapper script.

Reply:
Added a check for `LAUNCHER_PATH` like we do for the version check.